rust/storage/src/admissioncontrolleds3.rs

```diff
@@ -233,7 +233,7 @@ impl AdmissionControlledS3Storage {
             Range<usize>,
         ) -> BoxFuture<'static, Result<ByteStream, StorageError>>,
         options: PutOptions,
-    ) -> Result<(), StorageError> {
+    ) -> Result<Option<ETag>, StorageError> {
```

Comment:
Is the only reason this is Option to support local? It would be nice if it wasn't optional